packages/cursorless-engine/src/compositionRoot/createCursorlessEngine.ts

```diff
@@ -0,0 +1,59 @@
+import { CommandServerApi, Hats, IDE } from "@cursorless/common";
+import { CommandRunner, TestCaseRecorder, ThatMark, TreeSitter } from "..";
+import { Actions } from "../actions/Actions";
+import { Debug } from "../core/Debug";
+import { HatTokenMapImpl } from "../core/HatTokenMapImpl";
+import { Snippets } from "../core/Snippets";
+import { RangeUpdater } from "../core/updateSelections/RangeUpdater";
+import { injectIde } from "../singletons/ide.singleton";
+
+export function createCursorlessEngine(
+  treeSitter: TreeSitter,
+  ide: IDE,
+  hats: Hats,
+  commandServerApi: CommandServerApi | null,
```

Comment:
These parameters indicate pretty clearly what's required from any given instantiation of the Cursorless engine I think